applications/GeoMechanicsApplication/tests/cpp_tests/test_utilities/geo_custom_condition.h

```diff
@@ -0,0 +1,70 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Aron Noordam
+//
+#pragma once
+
+#include "includes/condition.h"
+
+namespace Kratos::Testing
+{
+
+class GeoCustomCondition : public Condition
+{
+public:
+    GeoCustomCondition() : Condition() {};
+    /// Constructor using Geometry
+    GeoCustomCondition(IndexType NewId, GeometryType::Pointer pGeometry)
+        : Condition(NewId, pGeometry) {};
+
+    GeoCustomCondition(IndexType NewId, const NodesArrayType& ThisNodes)
+        : Condition(NewId, std::make_shared<GeometryType>(ThisNodes))
+    {
+    }
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    Pointer Create(IndexType NewId, NodesArrayType const& ThisNodes, PropertiesType::Pointer pProperties) const override
+    {
+        return Kratos::make_intrusive<GeoCustomCondition>(NewId, GetGeometry().Create(ThisNodes));
+    }
+
+    Pointer Create(IndexType NewId, GeometryType::Pointer pGeom, PropertiesType::Pointer pProperties) const override
+    {
+        return Kratos::make_intrusive<GeoCustomCondition>(NewId, pGeom);
+    }
```

Comment:
If we don't go through the registration system, we can remove quite a few of these constructors/creates and only keep the things we actually use in our test